src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java

```diff
@@ -134,21 +131,45 @@ public static void logOnLowLevel(String command, String description, boolean suc
     log(command, description, success, true);
   }
 
-  private static void log(String command, String description, boolean isSuccess, boolean ifLowLevel) {
+  private static void log(String command, String description, boolean isSuccess,
+                          boolean ifLowLevel) {
     logsResults.add(isSuccess);
     String escapedDescription = escapeHtml(description);
 
-    String className = isSuccess ? "success" : "error";
-    StringBuilder builder = new StringBuilder();
+    String className = isSuccess ? SUCCESS_CLASS : ERROR_CLASS;
+    List<String> classList = new ArrayList<>();
+    classList.add(className);
+
     if (ifLowLevel) {
-      builder.append("<tr class=\"" + className + " lowLevelAction" + "\"><td>" + command
-                     + "</td><td>" + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
-    } else {
-      builder.append("<tr class=\"" + className + "\"><td>" + command + "</td><td>"
-                     + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
+      classList.add(LOW_LEVEL_ACTION_CLASS);
+    }
+    String html = VelocityWrapper.fillLogRow(classList, command, escapedDescription);
+    CommonUtils.appendTextToFile(logPath, html);
+    logJSError();
+  }
+
+  public void logAssertionStacktrace(AssertionError exception) {
+    driver = DriverProvider.getActiveDriver();
+
+    imageCounter += 1;
+    if ("true".equals(Configuration.getLogEnabled())) {
+      String exceptionMessage = ExceptionUtils.getStackTrace(exception);
+      List<String> classList = new ArrayList<>();
+      classList.add(ERROR_CLASS);
+      classList.add(STACKTRACE_CLASS);
+      String html = VelocityWrapper.fillErrorLogRow(classList, exceptionMessage, imageCounter);
+      try {
+        new Shooter().savePageScreenshot(screenPath + imageCounter, driver);
+        CommonUtils.appendTextToFile(screenPath + imageCounter + ".html", getPageSource(driver));
+        CommonUtils.appendTextToFile(logPath, html);
+      } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)